synapse/events/utils.py

```diff
@@ -100,22 +100,18 @@ def format_event_raw(d):
 
 
 def format_event_for_client_v1(d):
-    d["user_id"] = d.pop("sender", None)
+    d = format_event_for_client_v2(d)
+
+    d["user_id"] = d.get("sender", None)
```

Comment:
I dunno. I don't think this changes the behaviour of the code from before. I don't think we should be seeing events without a sender either. I'm happy to switch this to a conditional if you prefer.